File: packages/contract-npm/src/completion.ts
export enum EmitCompletionLevel {
  ACCEPTED_BY_RUNTIME = "accepted_by_runtime",
  QUEUED_FOR_TRANSPORT = "queued_for_transport",
  FLUSHED_TO_TRANSPORT = "flushed_to_transport",
  PEER_ACKNOWLEDGED = "peer_acknowledged",
  FAILED_DURING_EMIT = "failed_during_emit",
  ABORTED_BY_PEER = "aborted_by_peer",
}

File: packages/contract-rs/src/completion.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum EmitCompletionLevel {
    #[serde(rename = "accepted_by_runtime")]
    AcceptedByRuntime,
    #[serde(rename = "queued_for_transport")]
    QueuedForTransport,
    #[serde(rename = "flushed_to_transport")]
    FlushedToTransport,
    #[serde(rename = "peer_acknowledged")]
    PeerAcknowledged,
    #[serde(rename = "failed_during_emit")]
    FailedDuringEmit,
    #[serde(rename = "aborted_by_peer")]
    AbortedByPeer,
}
